sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/HttpTimeoutPolicy.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.http;
+
+import com.azure.cosmos.implementation.OperationType;
+import com.azure.cosmos.implementation.ResourceType;
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.RxDocumentServiceResponse;
+import io.netty.handler.codec.http.HttpMethod;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class HttpTimeoutPolicy {
+    final public static HttpTimeoutPolicy getTimeoutPolicy(RxDocumentServiceRequest request) {
+
+        if (OperationType.QueryPlan.equals(request.getOperationType()) ||
+            request.isAddressRefresh() ||
+            request.getResourceType() == ResourceType.PartitionKeyRange) {
+            return HttpTimeoutPolicyControlPlaneHotPath.instance;
+        }
+
+        return HttpTimeoutPolicyDefault.instance;
+    }
+
+    public abstract Duration maximumRetryTimeLimit();
+
+    public abstract Integer totalRetryCount();
```

Comment:
Can this ever be null? If not, better to use primitive type `int` here.

Reply:
I see all implementations of `totalRetryCount()` are the same -> 

```
@Override
    public Integer totalRetryCount() {
        return getTimeoutAndDelays().size();
    }
```

Also, all implementations of `getTimeoutList()` are also the same -> 

```
@Override
    public List<ResponseTimeoutAndDelays> getTimeoutList() {
        return getTimeoutAndDelays();
    }
```
Can we have a default implementation of `totalRetryCount()` as this?
```
public int totalRetryCount() {
        return getTimeoutList().size();
    }
```
and remove all the overrides to make it simple and clear.

Reply:
Added code to remove the overrides, and used primitive type int for totalRetryCount() method.